theme/templates/resource-landing-page/title-section.html

```diff
@@ -307,58 +309,69 @@ <h2 id="resource-title">{{ title }}</h2>
-    "keywords": "{{ keywords | escapejs }}",
-    {% if cm.metadata.rights.statement == "This resource is shared under the Creative Commons Attribution CC BY." %}
-        "license": "http://creativecommons.org/licenses/by/4.0/",
-    {% elif cm.metadata.rights.statement == "This resource is shared under the Creative Commons Attribution-ShareAlike CC BY-SA." %}
-        "license": "http://creativecommons.org/licenses/by-sa/4.0/",
-    {% elif cm.metadata.rights.statement == "This resource is shared under the Creative Commons Attribution-NoDerivs CC BY-ND." %}
-        "license": "http://creativecommons.org/licenses/by-nd/4.0/",
-    {% elif cm.metadata.rights.statement == "This resource is shared under the Creative Commons Attribution-NoCommercial-ShareAlike CC BY-NC-SA." %}
-        "license": "http://creativecommons.org/licenses/by-nc-sa/4.0/",
-    {% elif cm.metadata.rights.statement == "This resource is shared under the Creative Commons Attribution-NoCommercial CC BY-NC." %}
-        "license": "http://creativecommons.org/licenses/by-nc/4.0/",
-    {% elif cm.metadata.rights.statement == "This resource is shared under the Creative Commons Attribution-NoCommercial-NoDerivs CC BY-NC-ND." %}
-    "license": "http://creativecommons.org/licenses/by-nc-nd/4.0/",
-    {% endif %}
+    "keywords": {{ keywords|safe }},
     {% if cm.raccess.published %}
+    "creativeWorkStatus": "Published",
+    {% elif cm.raccess.public %}
+    "creativeWorkStatus": "Public",
+    {% elif cm.raccess.discoverable %}
+    "creativeWorkStatus": "Discoverable",
+    {% elif is_replaced_by %}
+    "creativeWorkStatus": "Obsolete",
+    {% else %}
+    "creativeWorkStatus": "Private",
+    {% endif %}
+    "inLanguage": "en-US",
     "identifier": {
-        "@type": "PropertyValue",
-        "additionalType": ["http://schema.geolink.org/1.0/base/main#Identifier", "http://purl.org/spar/datacite/Identifier"],
-        "propertyID": "http://purl.org/spar/datacite/doi",
+        "@type": [
+            "PropertyValue",
+            "datacite:ResourceIdentifier"
+        ],
+        {% if cm.raccess.published %}
+        "datacite:usesIdentifierScheme": {
+            "@id": "datacite:doi"
+        },
+        "propertyId": "DOI",
         "url": "https://doi.org/10.4211/hs.{{ cm.short_id }}",
         "value": "10.4211/hs.{{ cm.short_id }}"
+        {% else %}
+        "datacite:usesIdentifierScheme": {
+            "@id": "datacite:local-resource-identifier-scheme"
+        },
+        "propertyId": "UUID",
+        "value": "uuid:{{ cm.short_id }}"
+        {% endif %}
     },
-    {% endif %}
-    "citation": "{{ citation }}",
-    "includedInDataCatalog": {
-        "@id": "https://www.hydroshare.org"
-    },
-    "distribution": {
-        "@type": "DataDownload",
-        "contentUrl": "https://www.hydroshare.org/hsapi/resource/{{ cm.short_id }}/",
-        "encodingFormat": "application/zip"
-    },
+    "creator": {{ creators | creator_json_ld_element | safe }},
```

Comment:
I don't like that we're creating the json in two different ways.  I don't feel like it's anymore readable by building creators with data structures and dumping it to json  over building it here in the template.  And since the rest of the json is being built here, I'm thinking it makes sense to just do it here.

Reply:
That was what I initially did. If you look at how complicated the logic for creating creators json ld element is, you will understand why I moved it to a template filter tag. It will render template extremely unreadable if I move all the syntax handling here in the template rather than doing the filtering using backend support and then used json dumps function to create a json formatted string to be used directly by template. I think that is exactly what template filters are built for to handle complicated logic rather than doing it directly on template, right?